packages/flutter/test/widgets/interactive_viewer_test.dart

```diff
@@ -623,6 +623,43 @@ void main() {
       expect(transformationController.value.getMaxScaleOnAxis(), equals(1.0));
     });
 
+
+    testWidgets('Scale with mouse returns onInteraction properties',(WidgetTester tester) async{
+      final TransformationController transformationController = TransformationController();
+      double currentScale;
+      Velocity currentVelocity;
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: Center(
+              child: InteractiveViewer(
+                transformationController: transformationController,
+                onInteractionStart: (ScaleStartDetails details){
+
+                },
+                onInteractionUpdate: (ScaleUpdateDetails details){
+                  currentScale = details.scale;
+                },
+                onInteractionEnd: (ScaleEndDetails details){
+                  currentVelocity = details.velocity;
+                },
+                child: Container(width: 200.0, height: 200.0),
+              ),
+            ),
+          ),
+        ),
+      );
+
+      final Offset center = tester.getCenter(find.byType(InteractiveViewer));
+      await scrollAt(center, tester, const Offset(0.0, -20.0));
+      await tester.pumpAndSettle();
+      const Velocity noMovement = Velocity(pixelsPerSecond: Offset(0,0));
+      final double afterScaling = transformationController.value.getMaxScaleOnAxis();
+
+      expect(afterScaling,equals(currentScale));
+      expect(currentVelocity,equals(noMovement));
```

Comment:
```suggestion
      expect(afterScaling, equals(currentScale));
      expect(currentVelocity, equals(noMovement));
```

Reply:
Could you also expect that currentScale is not null and not 1.0?  I worry that this feature could break but the test would still pass if currentScale were 1.0 or null.